opengever/maintenance/scripts/fix_mail_class_query.py

```diff
@@ -0,0 +1,84 @@
+"""
+A bin/instance run script which checks and fixes the class of
+all mail objects based on querying the catalog.
+
+bin/instance0 run fix_mail_class_query.py
+"""
+
+from Acquisition import aq_base
+from Acquisition import aq_parent
+from ftw.mail.mail import Mail
+from opengever.mail.mail import OGMail
+from opengever.maintenance.debughelpers import setup_app
+from opengever.maintenance.debughelpers import setup_plone
+from Products.BTreeFolder2.BTreeFolder2 import BTreeFolder2Base
+from zope.app.intid.interfaces import IIntIds
+from zope.component import getUtility
+from zope.interface import directlyProvidedBy
+from zope.interface import directlyProvides
+import transaction
+
+
+def fix_mail_class(plone):
+    catalog = plone.portal_catalog
+    intids = getUtility(IIntIds)
+
+    counter = 0
+    mails_to_fix = []
+
+    mails = catalog.unrestrictedSearchResults(
+        {"portal_type": 'ftw.mail.mail'})
+
+    for brain in mails:
+        mail = brain.getObject()
+        if mail.__class__ == Mail:
```

Comment:
Hmm. Thinking about `__class__ == Mail` vs. `__class__ != OGMail`. In fact, I think I'd even prefer `__class__.__name__ != 'OGMail'` - then we could absolutely rule out the check failing because of different versions of the persisted class.